### ISSUE TITLE
Vagrant up fails with default: Warning: Authentication failure. Retrying... on Fedora 35

### ISSUE THREAD

**Issue by ericzolf** [hashicorp/vagrant#12749]
Please note that the Vagrant issue tracker is in priority reserved for bug reports and enhancements. For general usage questions, please use
HashiCorp Discuss: https://discuss.hashicorp.com/c/vagrant/24 Thank you!

### Vagrant version

2.2.16

### Host operating system

Fedora 35

### Guest operating system

centos/8 resp. centos/8stream

### Vagrantfile

It's just an unmodified Vagrantfile as generated by `vagrant init centos/8stream`

```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "centos/stream8"
end
```

### Debug output

https://gist.github.com/ericzolf/0c4aaca443a7f041fe8b42abc18756fb

### Expected behavior

The SSH connection should have worked.

### Actual behavior

The SSH connection fails again and again with `default: Warning: Authentication failure. Retrying...` and `vagrant up` goes into timeout.

This said, while waiting for the timeout to happen, the following works (that's why I think that it's an issue with Vagrant):

```
vagrant ssh-config > /tmp/ssh_config
ssh -F /tmp/ssh_config default
```

The content is:

```
$ cat /tmp/ssh_config
Host default
  HostName 192.168.121.51
  User vagrant
  Port 22
  UserKnownHostsFile /dev/null
  StrictHostKeyChecking no
  PasswordAuthentication no
  IdentityFile /home/ericl/.vagrant.d/insecure_private_key
  IdentitiesOnly yes
  LogLevel FATAL
```

### Steps to reproduce

1. vagrant init centos/8stream
2. vagrant up

### References

- GH issue #12000 (closed because the initial reporter couldn't reproduce the issue)

**Comment by ericzolf** [comment]
I tried with archlinux/archlinux as guest and get the same behavior.

**Comment by vinceskahan** [comment]
also happens in a ubuntu2204 vm for me...

**Comment by dt0vprizepicks** [comment]
Using `generic/ubuntu2204` the problem occurs for me too.

**Comment by theabro** [comment]
I also had this issue with an Ubuntu 20.04 host trying to provision an Ubuntu 22.04 guest. The base Vagrant version on 20.04 was 2.2.6. Upgrading to 2.3.4 from the Hashicorp repo has allowed the guest to be booted.

**Comment by chrisroberts** [comment]
Hi there,

This has been fixed via #13179 and can be resolved by upgrading your Vagrant installation.

Cheers!
